datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/MonthDeserializer.java

```diff
@@ -0,0 +1,48 @@
+package com.fasterxml.jackson.datatype.jsr310.deser;
+
+import java.io.IOException;
+import java.time.Month;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.deser.std.DelegatingDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+public class MonthDeserializer extends DelegatingDeserializer {
+    private final boolean _enableOneBaseMonths;
+
+    public MonthDeserializer(JsonDeserializer<Enum> defaultDeserializer, boolean oneBaseMonths) {
+        super(defaultDeserializer);
+        _enableOneBaseMonths = oneBaseMonths;
+    }
+
+    @Override
+    public Object deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        Month zeroBaseMonth = (Month) getDelegatee().deserialize(parser, context);
+        if (!_enableOneBaseMonths || zeroBaseMonth == null) {
+            return zeroBaseMonth;
+        }
+
+        JsonToken token = parser.currentToken();
+        if (token == JsonToken.VALUE_NUMBER_INT || _isNumberAsString(parser.getText(), token)) {
+            if (zeroBaseMonth == Month.JANUARY) {
+                throw new InvalidFormatException(parser, "Month.JANUARY value not allowed for 1-based Month.", zeroBaseMonth, Month.class);
+            } else {
+                return zeroBaseMonth.minus(1);
+            }
+        }
+        return zeroBaseMonth;
+    }
+
+    private boolean _isNumberAsString(String text, JsonToken token) throws IOException {
+        String regex = "[\"']?\\d{1,2}[\"']?";
```

Comment:
Ok, couple of things: `parser.getText()` will return contents, so no quotes around it, can leave those out.

And make sure to create pre-compiled `Pattern` as static member of class: otherwise RegEx state machine is re-create for every call... and that's pretty expensive (and more importantly easily avoidable :) ).